sysdig/resource_sysdig_secure_cloud_auth_account.go

```diff
@@ -387,25 +388,28 @@ func constructAccountComponents(data *schema.ResourceData) []*cloudauth.AccountC
 					if data.Get(SchemaCloudProviderType).(string) == cloudauth.Provider_PROVIDER_GCP.String() {
 						spGcp := &internalServicePrincipalMetadata{}
 						err = json.Unmarshal([]byte(value.(string)), spGcp)
+						// if GCP service principal key is present, decode and unmarshal it before populating metadata
+						var spGcpKey *cloudauth.ServicePrincipalMetadata_GCP_Key
 						if len(spGcp.Gcp.Key) > 0 {
+							log.Printf("SP component has a key...")
 							var spGcpKeyBytes []byte
 							spGcpKeyBytes, err = base64.StdEncoding.DecodeString(spGcp.Gcp.Key)
 							if err != nil {
 								diag.FromErr(err)
 							}
-							spGcpKey := &cloudauth.ServicePrincipalMetadata_GCP_Key{}
-							err = json.Unmarshal(spGcpKeyBytes, spGcpKey)
-							component.Metadata = &cloudauth.AccountComponent_ServicePrincipalMetadata{
-								ServicePrincipalMetadata: &cloudauth.ServicePrincipalMetadata{
-									Provider: &cloudauth.ServicePrincipalMetadata_Gcp{
-										Gcp: &cloudauth.ServicePrincipalMetadata_GCP{
-											Key:                        spGcpKey,
-											WorkloadIdentityFederation: spGcp.Gcp.WorkloadIdentityFederation,
-											Email:                      spGcp.Gcp.Email,
-										},
+							err = json.Unmarshal(spGcpKeyBytes, &spGcpKey)
+						}
+						log.Printf("Setting key: %v ; wif: %v ; email: %v", spGcpKey, spGcp.Gcp.WorkloadIdentityFederation, spGcp.Gcp.Email)
```

Comment:
Do you want to log the key here?

Reply:
Oops, thanks for pointing this out. Fixing it in next commit.